exporter/splunkhecexporter/client.go

```diff
@@ -260,20 +269,24 @@ func (c *client) fillMetricsBuffer(metrics pmetric.Metrics, buf buffer, is iterS
 
 				// Parsing metric record to Splunk event.
 				events := mapMetricToSplunkEvent(rm.Resource(), metric, c.config, c.logger)
-				tempBuf.Reset()
+
 				for _, event := range events {
 					// JSON encoding event and writing to buffer.
-					b, err := marshalEvent(event, c.config.MaxEventSize, jsonStream)
-					if err != nil {
-						permanentErrors = append(permanentErrors, consumererror.NewPermanent(fmt.Errorf("dropped metric event: %v, error: %w", event, err)))
+					jsonStream.Error = nil
+					jsonStream.WriteVal(event)
```

Comment:
Q.: this is the only `jsonStream.WriteVal(event)` that is without a corresponding `jsonStream.Flush()` - the flush in this case is outside the loop. Is that intentional? If yes, is it for performance reasons?

Reply:
It's because we have a case there where one metric maps to multiple Splunk events (one event per MTS, and Splunk converts histograms to multiple MTS). If one or more events cannot be written, we will need to retry the whole set to make sure we get all the data points over. The work on exporterhelper will allow us to do better there, but for now we keep the index of the metric we're at to retry.